File: trip-service-kata/scala/trip_service_kata/src/main/scala/com/dubswcraft/tripservicekata/user/UserSession.scala
package org.craftedsw.tripservicekata.user

import com.dubswcraft.tripservicekata.exception.CollaboratorCallException
import com.dubswcraft.tripservicekata.user.User

object UserSession {

	def getLoggedUser(): User = {
		throw new CollaboratorCallException(
			"UserSession.getLoggedUser() should not be called in an unit test");
	}

}

File: trip-service-kata/java/trip-service-kata/src/main/java/com/dubswcraft/tripservicekata/user/UserSession.java
package com.dubswcraft.tripservicekata.user;

public class UserSession {

	private static final UserSession userSession = new UserSession();
	
	private UserSession() {
	}
	
	public static UserSession getInstance() {
		return userSession;
	}

	public User getLoggedUser() {
		return new User("LOGGED_USER");
	}

}
